extension/src/openvic-extension/singletons/PopulationMenu.cpp

```diff
@@ -513,8 +510,16 @@ MenuSingleton::sort_func_t MenuSingleton::_get_population_menu_sort_func(populat
 	}
 }
 
-void MenuSingleton::_population_menu_sort_pops() {
+bool MenuSingleton::_population_menu_sort_pops() {
 	if (population_menu.sort_key != population_menu_t::NONE) {
+		if (
+			!population_menu.pop_type_sort_cache.has_keys() || !population_menu.culture_sort_cache.has_keys() ||
+			!population_menu.religion_sort_cache.has_keys() || !population_menu.province_sort_cache.has_keys() ||
+			!population_menu.rebel_type_sort_cache.has_keys()
+		) {
+			ERR_FAIL_COND_V(population_menu_update_locale_sort_cache() != OK, false);
```

Comment:
What is ERR_FAIL_COND_V?
Does it conditionally return false or throw?
Is there even a path that returns false in this method?

Sorry if this is obvious to you, macros often confuse me.

Reply:
This a godot-cpp assert macro that does not crash, if the evaluation is true, it returns the second value.

(ERR_FAIL_COND_V)[https://github.com/godotengine/godot-cpp/blob/f2b521f55a0a2ba7080418ba898b97028da6134a/include/godot_cpp/core/error_macros.hpp#L418]

Reply:
As Spartan said, if the first argument evaluates to false then the second argument is returned. We also use other similar macros from `godot-cpp` like `ERR_FAIL_NULL_V`, which returns its second argument if the first argument is false.